### ISSUE TITLE
[Bug]: `react/no-unused-prop-types` reports false positives if multiple components use the same prop type

### ISSUE THREAD

**Issue by ej612** [jsx-eslint/eslint-plugin-react#3868]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

I have 2 components that both share the same prop type. The first component doesn't use all props, the second does. The rule reports an error on the second prop claiming it's unused (since it's not used by the first component). It seems that the rule only looks at the first component it can find and doesn't take further components into consideration:
```tsx
import React from 'react';

interface Props {
    prop1: string,

    // False positive here:
    // 'prop2' PropType is defined but prop is never used eslint(react/no-unused-prop-types)
    prop2: boolean
}

// this component doesn't use prop2
export const Component1 = (props: Props) => {
    return <div title={props.prop1}/>;
};

// this component does
export const Component2 = (props: Props) => {
    return <div title={props.prop1} aria-disabled={props.prop2}/>;
};
```

### Expected Behavior

I would expect the rule to look at all components that use a certain prop type instead of only the first one it can find.

### eslint-plugin-react version

7.37.2

### eslint version

9.16.0

### node version

18.20.5

**Comment by ljharb** [comment]
It’s a great idea to share props between components - but only when they have the same signature.

It’s objectively and inarguably a bad idea to do what being done here, and the rule shouldn’t allow you to do it - so it’s behaving correctly.
